projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -508,990 +324,646 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
+            this.dateTimeEditor.clear();
         }
     }
 
     /**
-     * Opens the date picker drop down or dialog.
+     * Increment a specified `DatePart`.
      *
-     * @param target HTMLElement - the target element to use for positioning the drop down container according to
+     * @param datePart The optional DatePart to increment. Defaults to Date.
+     * @param delta The optional delta to increment by. Overrides `spinDelta`.
      * @example
      * ```typescript
-     * this.datePicker.openDialog(target);
+     * this.datePicker.increment(DatePart.Date);
      * ```
      */
-    public openDialog(): void {
-        if (!this.collapsed || this.disabled) {
-            return;
-        }
-
-        switch (this.mode) {
-            case InteractionMode.Dialog: {
-                this.hasHeader = true;
-                const modalOverlay = (this.modalOverlaySettings !== undefined) ? this._modalOverlay : this._modalOverlaySettings;
-                this._componentID = this._overlayService.attach(IgxCalendarContainerComponent, modalOverlay, this._moduleRef);
-                this._overlayService.show(this._componentID);
-                break;
-            }
-            case InteractionMode.DropDown: {
-                this.hasHeader = false;
-                const target = this.inputGroupElement;
-                if (target) {
-                    this.dropDownOverlaySettings.target = target;
-                }
-                this._componentID = this._overlayService.attach(IgxCalendarContainerComponent,
-                    this.dropDownOverlaySettings, this._moduleRef);
-                this._overlayService.show(this._componentID);
-                break;
-            }
-        }
+    public increment(datePart?: DatePart, delta?: number): void {
+        this.dateTimeEditor.increment(datePart, delta);
     }
 
     /**
-     * Close the calendar.
+     * Decrement a specified `DatePart`
      *
-     * @hidden @internal
+     * @param datePart The optional DatePart to decrement. Defaults to Date.
+     * @param delta The optional delta to decrement by. Overrides `spinDelta`.
+     * @example
+     * ```typescript
+     * this.datePicker.decrement(DatePart.Date);
+     * ```
      */
-    public closeCalendar(): void {
-        this._overlayService.hide(this._componentID);
+    public decrement(datePart?: DatePart, delta?: number): void {
+        this.dateTimeEditor.decrement(datePart, delta);
     }
 
-    /**
-     * Clear the input field, date picker value and calendar selection.
-     *
-     * @hidden @internal
-     */
-    public clear(): void {
-        if (!this.disabled) {
-            this.isEmpty = true;
-            this.invalidDate = '';
-            this.deselectDate();
-            this._setCursorPosition(0);
+    //#region Control Value Accessor
+    /** @hidden @internal */
+    public writeValue(value: Date) {
+        this._value = value;
+        if (this.dateTimeEditor.value !== value) {
```

Comment:
Is this not bound already?

Reply:
No it's now bound, here we only update the internal `_value` and we also have to tell the directive to update its own value as well.